packages/vsx-registry/src/browser/vsx-extensions-contribution.ts

```diff
@@ -215,11 +242,66 @@ export class VSXExtensionsContribution extends AbstractViewContribution<VSXExten
         }
     }
 
+    /**
+     * Given an extension, displays a quick pick of other compatible versions and installs the selected version.
+     *
+     * @param extension a VSX extension.
+     */
+    protected async installAnotherVersion(extension: VSXExtension): Promise<void> {
+        const extensionId = extension.id;
+        const currentVersion = extension.version;
+        const extensions = await this.vsxRegistryAPI.getAllVersions(extensionId);
+        const latestCompatible = await this.vsxRegistryAPI.getLatestCompatibleExtensionVersion(extensionId);
+        let compatibleExtensions: VSXExtensionRaw[] = [];
+        if (latestCompatible) {
+            compatibleExtensions = extensions.slice(extensions.findIndex(ext => ext.version === latestCompatible.version));
+        }
+        const items: QuickPickItem<QuickPickVersionItem>[] = [];
+        compatibleExtensions.forEach(ext => {
+            let publishedDate = moment(ext.timestamp).fromNow();
+            if (currentVersion === ext.version) {
+                publishedDate += ' (Current)';
+            }
+            items.push({
+                label: ext.version,
+                value: { version: ext.version },
+                description: publishedDate
+            });
+        });
+        const selectedItem = await this.quickPickService.show(items, { placeholder: 'Select Version to Install', runIfSingle: false });
+        if (selectedItem && currentVersion && selectedItem.version !== currentVersion) {
+            const selectedExtension = this.model.getExtension(extensionId);
+            if (selectedExtension) {
+                await this.updateVersion(selectedExtension, selectedItem.version, currentVersion);
+            }
+        }
+    }
+
     protected async copy(extension: VSXExtension): Promise<void> {
         this.clipboardService.writeText(await extension.serialize());
     }
 
     protected copyExtensionId(extension: VSXExtension): void {
         this.clipboardService.writeText(extension.id);
     }
+
+    /**
+     * Updates an extension to a specific version.
+     *
+     * @param extension the extension to update.
+     * @param updateToVersion the version to update to.
+     * @param revertToVersion the version to revert to (in case of failure).
+     */
+    private async updateVersion(extension: VSXExtension, updateToVersion: string, revertToVersion: string): Promise<void> {
```

Comment:
Maybe some message service toasts here if the process fails at any step where the user might want to know? I'm not 100% sure about what messages may be displayed by the functions called here, but it seems like the user might want to know if they ended up with the old version again, for example.